PHPMailer/PHPMailer
# Tests: move tests for the various mail transports to separate file

This is a next PR in a series of PRs to improve the test suite and make it easier to maintain.
The principle of these changes was proposed to and discussed with the maintainer prior to work being started on it.

I'm splitting this up into several PRs to 1) allow for easier review and 2) allow the repo to benefit from the changes as quickly as possible. (the complete series is still a WIP).

Previous PRs in this series: #2372, #2376, #2377, #2378, #2379

---

👉🏻 **_Note: I hope I caught all the relevant tests for this test file. Please let me know if there are any other tests which should be moved to the `MailTransportTest` class._**

## Commit details

### Tests/reorganize: move email transport tests to own file

### MailTransportTest: various test tweaks

Minor test tweaks:
* Add `@covers` tags ( 👉🏻 **Needs review! - especially the `testMailSend()` method seems to do more than it should**)
* Check if test skipping is necessary at the start of a test method.
* Add "failure message" for each assertion in tests with multiple assertions.
* Tidy up inline comments.
